tests/Authentication/ForcePasswordResetTest.php

```diff
@@ -0,0 +1,112 @@
+use Config\Services;
+use Tests\Support\TestCase;
+
+/**
+ * @internal
+ */
+final class ForcePasswordResetTest extends TestCase
+{
+    use DatabaseTestTrait;
+    use FeatureTestTrait;
+    use AuthenticationTesting;
+
+    protected $refresh = true;
+    protected $namespace;
+    private string $routeFilter = 'force-reset';
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        // Load our Auth routes in the collection
+        $routeCollection = service('routes');
+        auth()->routes($routeCollection);
+
+        $routeCollection->get('profile', static function (): void {
+            echo 'Profile';
+        }, ['filter' => $this->routeFilter]);
+
+        Services::injectMock('routes', $routeCollection);
+    }
+
+    public function testUserRequiresPasswordReset(): void
+    {
+        /** @var User $user */
+        $user = fake(UserModel::class);
+        $user->createEmailIdentity(['email' => 'foo@example.com', 'password' => 'secret123']);
+
+        /** @var UserIdentityModel $identity */
+        $identity = model(UserIdentityModel::class);
+        $identity->set('force_reset', 1);
+        $identity->where('user_id', $user->id);
+        $identity->update();
+
+        $this->assertTrue($user->requiresPasswordReset());
+    }
+
+    public function testForcePasswordResetOnUser(): void
+    {
+        /** @var User $user */
+        $user = fake(UserModel::class);
+        $user->createEmailIdentity(['email' => 'foo@example.com', 'password' => 'secret123']);
+        $user->forcePasswordReset();
+
+        $this->assertTrue($user->requiresPasswordReset());
+    }
+
+    public function testUndoForcePasswordResetOnUser(): void
+    {
+        /** @var User $user */
+        $user = fake(UserModel::class);
+        $user->createEmailIdentity(['email' => 'foo@example.com', 'password' => 'secret123']);
+        $user->forcePasswordReset();
+        $user->undoForcePasswordReset();
+
+        $this->assertFalse($user->requiresPasswordReset());
+    }
+
+    public function testRequiresPasswordResetRedirect(): void
+    {
+        /** @var User $user */
+        $user = fake(UserModel::class);
+        $user->createEmailIdentity(['email' => 'foo@example.com', 'password' => 'secret123']);
+        $user->forcePasswordReset();
+
+        $result = $this->actingAs($user)->get('profile');
+
+        // This should redirect to the stated url in Auth redirects
+        $result->assertStatus(302);
+        $result->assertRedirectTo(config('Auth')->forcePasswordResetRedirect());
+    }
+
+    public function testForceGlobalPasswordReset(): void
+    {
+        /** @var User $user */
+        $user = fake(UserModel::class);
+        $user->createEmailIdentity(['email' => 'foo@example.com', 'password' => 'secret123']);
+        $this->actingAs($user);
+
+        /** @var Fabricator $fabricator */
+        $fabricator = new Fabricator(UserIdentityModel::class);
+        $fabricator->create(50);
```

Comment:
This data generation cannot guarantee that email addresses are unique.
It was by chance that the test passed.
```
1) Tests\Authentication\ForcePasswordResetTest::testForceGlobalPasswordReset
CodeIgniter\Database\Exceptions\DatabaseException: Duplicate entry 'tmraz' for key 'username'
```
https://github.com/codeigniter4/shield/actions/runs/4081141840/jobs/7034271398

I will fix it.

Reply:
Oh, that must have been an oversight, I had it in mind to add the `unique()` modifier, just like I did in the `UserModel`.

Thanks @kenjis.

Reply:
I sent PR #621